### ISSUE TITLE
Pygame 2.0.0dev6: display surface has size (1,1)

### ISSUE THREAD

**Issue by fladd** [pygame/pygame#1475]
On MacOS with Pygame 2, this

```
import pygame
pygame.init()
print(pygame.__version__)
screen = pygame.display.set_mode((400, 300), pygame.DOUBLEBUF | pygame.OPENGL)
print(screen.get_size())
```

will result in

```
2.0.0.dev6
(1, 1)
```

while in Pygame 1.9, it used to be

```
1.9.6
(400, 300)
```

**Comment by robertpfeiffer** [comment]
I'm not sure if relying on the content of the fake display surface is a good idea. Do any of your games rely on this behaviour? In the future we will provide a better API for this, but if there are games that rely on the size of the fake display surface being the window size, we could re-create the old behaviour.

**Comment by fladd** [comment]
Well, [we](expyriment.org) do, but we can rewrite our code.

Please be aware that also `pygame.display.get_surface().get_size()` and `(pygame.display.Info().current_w`/`(pygame.display.Info().current_h` will not return the actual size anymore in Pygame 2!

Only `pygame.display.get_window_size()` does return the correct size it seems.

**Comment by humanbeast7** [comment]
@robertpfeiffer 
well, the previous API was very convenient
beside that, using the fake display surface "get_size()"
might be called in a function that can be applied to any surface,
therefore, the new API will make every function like that
to use a special case for the display surface.
as an example, think of a function that checks collision with surface borders

**Comment by robertpfeiffer** [comment]
@fladd yeah.
`pygame.display.Info()` should actually be changed.

`pygame.display.get_window_size()` gives you the actual physical size

**Comment by fladd** [comment]
I just think that the fact that the display surface is a "fake" surface, while technically correct, is not something that concerns the user. For the user, conceptually the display has a surface they can blit on, and that surface, conceptually, is the size of the display/window.

**Comment by robertpfeiffer** [comment]
I understand where you're coming from, but no.

I'll fix this, but in general, the return value of `pygame.display.set_mode(spam, pygame.OPENGL)` is a fake surface, and *should not* be treated like a surface, and you *should not* blit to or from it. You *do not* blit to the screen with `pygame.OPENGL`. You use OpenGL instead of blitting. That's the whole point.

Likewise, when you do `pygame.display.set_mode(spam, bacon | pygame.SCALED)`, then your surface size won't necessarily be the size of the window.

I'd fix the behaviour (for backwards compat) but also change the docs that we make no guarantee about the type or behaviour of the return value when using `pygame.OPENGL`.

While I'm at it, I could also write a new is_fullscreen API for #1343 and deprecate the FULLSCREEN flag on surfaces *in the docs*.

**Comment by illume** [comment]
@fladd thanks for the issue report. It's fixed in #1482

Quite a few OPENGL using apps depend on this behaviour. So this behaviour is not going away.

**Comment by fladd** [comment]
> @fladd thanks for the issue report. It's fixed in #1482
> 
> Quite a few OPENGL using apps depend on this behaviour. So this behaviour is not going away.

Glad to hear this. Thanks for fixing!
